_posts/en/newsletters/2020-07-29-newsletter.md

```diff
@@ -0,0 +1,236 @@
+[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #19473][] adds support for `networkactive` as both a command line
+  start-up and configuration file option. Setting this option enables or
+  disables all P2P network activity. After the node has been started, network
+  activity can be toggled using the existing `setnetworkactive` RPC or the network
+  activity button in the GUI.
+
+- [Bitcoin Core #18044][] adds support for witness txid (wtxid)
+  transaction inventory announcements (`inv`) and requests (`getdata`)
+  as described in BIP339 (see [Newsletter #104][news104 bip339]).  Prior
+  to this merge, all Bitcoin nodes announced new unconfirmed
+  transactions to their peers by the transaction's txid.  However, txids
+  don't commit to the witness data in segwit transactions, so a node who
+  downloaded an invalid or unwanted segwit transaction can't safely
+  assume that any transaction with that same txid is also invalid or
+  unwanted.  That means nodes may waste bandwidth by downloading the
+  same bad transaction over and over from each peer announcing that
+  transaction.
+
+    So far this hasn't been an issue---honest peers usually don't
+    announce transactions they wouldn't accept themselves, so only a
+    disruptive peer that wanted to waste its own upload bandwidth would
+    advertise invalid or unwanted transactions.  However, one type of
+    unwanted transaction today are spends of v1 segwit UTXOs---the types
+    of spends the [BIP341][] specification of [taproot][topic taproot]
+    plans to use.  If taproot activates, this means newer taproot-aware
+    nodes will advertise taproot spends to older taproot-unaware nodes.
+    Each time one those taproot-unaware nodes receives a
+    taproot-spending transaction, it will download it, realize it uses
+    v1 segwit, and throw it away.  This could be very wasteful of
+    network bandwidth, both for older taproot-unaware nodes and newer
+    taproot-aware nodes.  This same problem applies to other proposed
+    changes to network relay policy.
+
+    The solution implemented in this merged PR is to announce
+    transactions by their wtxid---which includes a commitment to the
+    witness data for segwit transactions.  A taproot implementation in
+    Bitcoin Core (see [PR #17977][Bitcoin Core #17977]) could then only
+    relay transactions by their wtxid to prevent newer nodes from
+    accidentally spamming older nodes.
+
+    However, after this PR was merged into Bitcoin Core's master
+    development branch, it was [discussed][meeting xscript] during the
+    weekly Bitcoin Core Development Meeting whether taproot's soft
+    dependency on wtxid relay will make it more complicated to backport
+    taproot to the current 0.20.x branch of Bitcoin Core.  Four options
+    were mentioned during the meeting and in subsequent discussions:
```

Comment:
The latest suhas/sipa irc discussion at http://www.erisian.com.au/bitcoin-core-dev/log-2020-07-28.html#l-511 and http://www.erisian.com.au/bitcoin-core-dev/log-2020-07-29.html#l-3 may be relevant here.

Reply:
These are good, but I think they basically resolve into "backporting wtxid is best", so I'm going to leave readers just the meeting link.